src/null-safety/index.md

```diff
@@ -1,39 +1,247 @@
+(assuming you’ve opted into null safety):
+
+```dart
+int aRealInt; // Can't be used until it's initialized to an integer.
+```
+
+If the variable _can_ have the value `null`,
+**add  `?`** to its type declaration:
+
+```dart
+int? aNullableInt = null;
+```
+
+When using a value that you know isn’t null but that has a nullable type,
+**add `!`** to tell the Dart analyzer that the value isn’t null:
+
+```dart
+int? a = 2;
+int b = a!; // The ! temporarily changes `a` from int? to int.
+```
+
+If you know that a non-nullable value will be
+initialized to a non-null value before it's used,
+but the Dart analyzer thinks otherwise,
+**add the `late` keyword** to the type:
+
+```dart
+late int aRealInt;
+```
+
+{% comment %}
+{{ site.alert.tip }}
+  Avoid using `late final` unless you really need it.
+  For example, don't use `late final` in functions,
+  and in constructors prefer using an [initializer list][].
+{{ site.alert.end }}
+
+[initializer list]: /guides/language/language-tour#initializer-list
+{% endcomment %}
+
+
+## Understanding list, set, and map types
+
+Lists, sets, and maps are commonly used collection types in Dart programs,
+so you need to know how they interact with null safety.
+Here are some examples of how Dart code uses these collection types:
+
+* Flutter layout widgets such as [`Column`][]
+  often have a `children` property that’s a
+  `List` of `Widget` objects.
+* The [Veggie Seasons][] sample uses a `Set` of `VeggieCategory` to
+  store a user’s food preferences.
+* The [GitHub Dataviz][] sample has a `fromJson()`method that
+  creates an object from JSON data that’s supplied in a
+  `Map<String, dynamic>`.
+
+[`Column`]: https://api.flutter.dev/flutter/widgets/Column-class.html
+[Veggie Seasons]: https://github.com/flutter/samples/tree/master/veggieseasons
+[GitHub Dataviz]: https://github.com/flutter/samples/tree/master/web/github_dataviz
+
+
+### List and set types {#list-and-set-types}
+
+When you’re declaring the type of a list or set, think about what can be null. The following table shows the possibilities for a list of strings
+if you opt into null safety.
+
+{% assign yes = '<b>Yes</b>' %}
+
+|------------------+----------------------------------+--------------------------+------------|
+|Type              | Can an item (string)<br>be null? | Can the list<br>be null? | Description|
+|------------------|----------------------------------|--------------------------|------------|
+| `List<String>`   | No      | No      | A non-null list that contains<br> non-null strings       |
+| `List<String>?`  | No      | {{yes}} | A non-null list that contains<br> strings that **might be null** |
+| `List<String?>`  | {{yes}} | No      | A list that **might be null** and that<br> contains non-null strings |
+| `List<String?>?` | {{yes}} | {{yes}} | A list that **might be null** and that<br> contains strings that **might be null**<br> (same as `List<String>` if you <br>don't opt into null safety) |
```

Comment:
The "same as" comment is confusing. It mixes old syntax with new syntax and it's not clear what is meant. 
Can be read as saying that "List<String?>?` is a non-nullable list of non-nullable strings if you don't opt into null safety. (That is, read as: "same as `List<String>` above if ..."). 

Just drop the parenthesized comment.

Reply:
I can remove it from the table, but it seems worth pointing out somewhere that the old `List<String>` would be expressed as `List<String?>?`. (Could be in a footnote.) Or do people not need to know/understand this?

Reply:
It's only partly true.

The old `List<String>` works like the new `List<String?>?` when assigning to it, but like the new `List<String>` when reading from it or calling methods. 

If you have `List<String?>? list; print(list.length);`, it's not valid. It would be for `List<String>` (both new and old interpretations).

I'm not against saying something, but it needs to be very, very clear when syntax is interpreted in the old way instead of the new way. I'm not sure what it's saying is actually worth the necessary effort.